src/screens/planActivity/PlanInput.tsx

```diff
@@ -1,16 +1,43 @@
-import React, { SFC } from 'react';
+import React from 'react';
 import { StyleSheet, View } from 'react-native';
 
 import { i18n } from 'locale';
+import { Plan } from 'models';
 import { palette } from 'styles';
 import { Icon, TextInput } from '../../components';
 
-export const PlanInput: SFC = () => (
-  <View style={styles.container}>
-    <Icon name="emoticon" size={24} color={palette.textInputPlaceholder} />
-    <TextInput style={styles.textInput} placeholder={i18n.t('planActivity:planNamePlaceholder')} />
-  </View>
-);
+interface Props {
+  studentId: string;
+}
+
+interface State {
+  text: string;
+}
+
+export class PlanInput extends React.PureComponent<Props, State> {
+  state: State = {
+    text: '',
+  };
+
+  handleBlur = () => Plan.create(this.props.studentId, this.state.text);
+
+  handleChangeText = (text: string) => this.setState({ text });
+
+  render() {
+    return (
+      <View style={styles.container}>
+        <Icon name="emoticon" size={24} color={palette.textInputPlaceholder} />
+        <TextInput
+          style={styles.textInput}
+          placeholder={i18n.t('planActivity:planNamePlaceholder')}
+          value={this.state.text}
+          onChangeText={this.handleChangeText}
+          onBlur={this.handleBlur}
```

Comment:
Isn't `onEndEditing` more suitable event for this case?

Reply:
You are right because, then callback that is called when text input ends.